src/wallpaper.rs

```diff
@@ -98,40 +101,57 @@ impl Wallpaper {
         let start = Instant::now();
         let mut cur_resized_img: Option<DynamicImage> = None;
 
-        for layer in self
+        // Only draw if all layers have been configured with size, and scale
+        if self
             .layers
-            .iter_mut()
-            .filter(|layer| !layer.first_configure)
+            .iter()
+            .any(|l| l.size.is_none() || l.fractional_scale.is_none())
         {
+            return;
+        }
+
+        for layer in self.layers.iter_mut().filter(|layer| layer.needs_redraw) {
             let Some(pool) = layer.pool.as_mut() else {
                 continue;
             };
 
-            if cur_resized_img.as_ref().map_or(true, |img| {
-                img.width() != layer.width || img.height() != layer.height
-            }) {
-                let CosmicBgLayer { width, height, .. } = *layer;
+            let fractional_scale = layer.fractional_scale.unwrap();
```

Comment:
So we require the fractional-scale protocol now?

Reply:
I'm assuming cosmic-bg isn't really targeting compositors other than cosmic-comp, so there's no harm in depending on protocols cosmic-comp provides. (Especially if it's a standard protocol other current compositors should support.)

We could definitely not require it, but it makes this a bit more complicated, particularly if we still want to support integer scaling without the protocol.